src/RA_Dlg_MemBookmark.cpp

```diff
@@ -704,50 +704,74 @@ void Dlg_MemBookmark::ExportJSON()
     defaultDir.erase(0, 2); // Removes the characters (".\\")
     defaultDir = g_sHomeDir + defaultDir;
 
-    IFileSaveDialog* pDlg = nullptr;
-    HRESULT hr = CoCreateInstance(CLSID_FileSaveDialog, nullptr, CLSCTX_ALL, IID_IFileSaveDialog, reinterpret_cast<void**>(&pDlg));
-    if (hr == S_OK)
+	CComPtr<IFileSaveDialog> pDlg;
+	
+	if (auto hr = CoCreateInstance(CLSID_FileSaveDialog, nullptr, CLSCTX_ALL, IID_IFileSaveDialog,
+        reinterpret_cast<void**>(&pDlg)); SUCCEEDED(hr))
     {
-        hr = pDlg->SetFileTypes(ARRAYSIZE(c_rgFileTypes), c_rgFileTypes);
-        if (hr == S_OK)
+		
+		if (hr = pDlg->SetFileTypes(c_rgFileTypes.size(), &c_rgFileTypes.front()); SUCCEEDED(hr))
         {
-            char defaultFileName[512];
-            sprintf_s(defaultFileName, 512, "%s-Bookmarks.txt", std::to_string(g_pCurrentGameData->GetGameID()).c_str());
-            hr = pDlg->SetFileName(Widen(defaultFileName).c_str());
-            if (hr == S_OK)
+            std::ostringstream oss;
+            oss.str().reserve(512); // storage on the heap really shouldn't matter but keeping relativelty the same
+            oss << g_pCurrentGameData->GetGameID() << "-Bookmarks.txt";
+
+            auto defaultFileName{ oss.str() };
+            // get in the habit of reseting the stringbuffer inside the
+            // stringstream if you wish to use it more than once (you will get
+            // garabage characters in the end)
+            oss.str(""); 
```

Comment:
Comment may be valid (despite the spelling mistakes), but this isn't used in a loop, so it's not relevant here and should be removed. If it is being used below (i didn't see it), the reset should happen there.

Reply:
OK, I'll just save those notes for myself and get the VS Spell Checker while I'm at it (it's an extension has nothing to do with code).